platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/TaskSchedulerConfiguration.java

```diff
@@ -0,0 +1,249 @@
+            final Boolean parsedFlushing = tryToParseFlushing(strippedPart);
+            if (parsedFlushing != null) {
+                if (flushingEnabled != null) {
+                    throw new IllegalStateException("Multiple flushing configurations specified: " + string);
+                }
+                flushingEnabled = parsedFlushing;
+                continue;
+            }
+
+            final Boolean parsedSquelching = tryToParseSquelching(strippedPart);
+            if (parsedSquelching != null) {
+                if (squelchingEnabled != null) {
+                    throw new IllegalStateException("Multiple squelching configurations specified: " + string);
+                }
+                squelchingEnabled = parsedSquelching;
+            }
+        }
+
+        return new TaskSchedulerConfiguration(
+                type,
+                unhandledTaskCapacity,
+                unhandledTaskMetricEnabled,
+                busyFractionMetricEnabled,
+                flushingEnabled,
+                squelchingEnabled);
+    }
+
+    /**
+     * Try to parse a string as a {@link TaskSchedulerType}.
+     *
+     * @param string the string to parse
+     * @return the parsed type, or null if the string is not a valid type
+     */
+    @Nullable
+    private static TaskSchedulerType tryToParseTaskSchedulerType(@NonNull final String string) {
+        try {
+            return TaskSchedulerType.valueOf(string);
+        } catch (final IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Try to parse a string as a capacity.
+     *
+     * @param string the string to parse
+     * @return the parsed capacity, or null if the string is not a valid capacity
+     */
+    @Nullable
+    private static Long tryToParseCapacity(@NonNull final String string) {
+        if (string.startsWith(TaskSchedulerConfigOption.CAPACITY)) {
+
+            try {
+                // parse a string in the form "CAPACITY(1234)"
+                final int openParenIndex = string.indexOf('(');
+                final int closeParenIndex = string.indexOf(')');
+                if (openParenIndex == -1 || closeParenIndex == -1) {
+                    throw new IllegalArgumentException("Invalid capacity \"" + string + "\"");
+                }
+                final String capacityString = string.substring(openParenIndex + 1, closeParenIndex);
+                return Long.parseLong(capacityString);
+            } catch (final NumberFormatException e) {
+                throw new IllegalArgumentException("Invalid capacity \"" + string + "\"", e);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Try to parse a string as an unhandled task metric configuration.
+     *
+     * @param string the string to parse
+     * @return the parsed configuration, or null if the string is not a valid configuration
+     */
+    private static Boolean tryToParseUnhandledTaskMetric(@NonNull final String string) {
```

Comment:
`tryToParseUnhandledTaskMetric`, `tryToParseBusyFractionMetric`, `tryToParseFlushing`, and `tryToParseSquelching` could all be replaced with a single method:

```
    private static Boolean tryToParseOption(
            @NonNull final String string, @NonNull final TaskSchedulerConfigOption option) {
        if (string.equals(option.toString())) {
            return true;
        } else if (string.equals("!" + option)) {
            return false;
        }
        return null;
    }
```